gitlab/gitlab_test.go

```diff
@@ -43,3 +50,104 @@ func TestGetProjectID(t *testing.T) {
 		t.Errorf("Expected %s, got %s", "1", res)
 	}
 }
+
+func TestGetProjectIDwithNonexistentProject(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+	mockURL := "https://" + "gitlab.com" + "/api/v4"
+	httpmock.RegisterResponder("GET", "https://gitlab.com/api/v4/projects/",
+		httpmock.NewStringResponder(404, ""))
+	_, err := GetProjectID(mockURL, "213123", "test", "test")
+	if err == nil {
+		t.Errorf("Expected to get an error when project does not exist")
+	}
+}
+
+func TestGitlabCreateAndDisplayNewMilestones(t *testing.T) {
+	milestoneData := utils.CreateMilestoneData(10, "daily", nil, "gitlab")
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+	mockURL := "https://" + "gitlab.com" + "/api/v4"
+	MockGitlabAPIGetRequest(mockURL, "active")
+	MockGitlabAPIPostRequest(mockURL, "active")
+	err := CreateAndDisplayNewMilestones(mockURL, "213123", "1", milestoneData, logger)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestPaginate(t *testing.T) {
```

Comment:
As with the Github paginate test, it feels like a test manufactured to succeed without actually testing the function.